apps/ledger-live-desktop/src/renderer/families/ethereum/MaxBaseFeeField.jsx

```diff
@@ -0,0 +1,142 @@
+  max-height: ${p => (p.hasError ? 60 : 0)}px;
+  min-height: ${p => (p.hasError ? 22 : 0)}px;
+  `;
+
+  
+const ErrorDisplay = styled(Box)`
+  color: ${p => p.theme.colors.pearl};
+`;
+
+const WarningDisplay = styled(Box)`
+  color: ${p => p.theme.colors.warning};
+`;
+
+type Props = {
+  account: Account,
+  transaction: Transaction,
+  status: TransactionStatus,
+  updateTransaction: (updater: any) => void,
+};
+
+let lastNetworkNextBaseFeePerGas;
+
+const FeesField = ({
+  account,
+  transaction,
+  status,
+  updateTransaction,
+  t
+}: Props & { t: TFunction }) => {
+  invariant(transaction.family === "ethereum", "FeeField: ethereum family expected");
+
+
+  const bridge = getAccountBridge(account);
+
+  const onMaxBaseFeeChange = useCallback(
+    maxBaseFee => {
+      updateTransaction(transaction =>
+        bridge.updateTransaction(transaction, { maxBaseFeePerGas: maxBaseFee, feesStrategy: "advanced" }),
+      );
+    },
+    [updateTransaction, bridge],
+  );
+
+  const networkNextBaseFeePerGas = transaction.networkInfo && transaction.networkInfo.nextBaseFeePerGas;
+  if ((!lastNetworkNextBaseFeePerGas && networkNextBaseFeePerGas)
+    || lastNetworkNextBaseFeePerGas !== networkNextBaseFeePerGas) {
+    lastNetworkNextBaseFeePerGas = networkNextBaseFeePerGas;
+  }
+  console.log('Now my1 transaction is (in maxBaseFee):', transaction)
+  const maxBaseFeePerGas = transaction.maxBaseFeePerGas || lastNetworkNextBaseFeePerGas;
+  const { units } = account.currency;
+  const unit = units.length > 1 ? units[1] : units[0];
+  const unitName = unit.code;
+
+  const validTransactionError = status.errors.maxBaseFee;
+  const validTransactionWarning = status.warnings.maxBaseFee;
+
+  return (
+    <Box mb={1}>
+      <LabelWithExternalIcon
+        onClick={() => {
+          openURL(urls.feesEIP1559MoreInfo);
+          track("Send Flow EIP1559 Fees Help Requested");
+        }}
+        label={t("send.steps.details.ethereumMaxBaseFee", { unitName })}
+      />
+      <Box mt={2} mb={2}>
+        <InputCurrency
+          hideErrorMessage
+          error={validTransactionError}
+          warning={validTransactionWarning}
+          containerProps={{ grow: true }}
+          defaultUnit={unit}
+          value={maxBaseFeePerGas}
+          onChange={onMaxBaseFeeChange}
+        />
+      </Box>
+      <ErrorContainer hasError={validTransactionError || validTransactionWarning}>
+        {validTransactionError ? (
+          <ErrorDisplay id="input-error">
+            <TranslatedError error={validTransactionError} />
+          </ErrorDisplay>
+        ) : validTransactionWarning ? (
+          <WarningDisplay id="input-warning">
+            <TranslatedError error={validTransactionWarning} />
+          </WarningDisplay>
+        ) : null}
+      </ErrorContainer>
+      <Label>
+        <span>
+          <Trans i18nKey="send.steps.details.nextBlock" values={{
+            nextBaseFee: formatCurrencyUnit(unit, lastNetworkNextBaseFeePerGas,
```

Comment:
consider a `useMemo` to store the result of formatCurrencyUnit because it can be unefficient if there are lot of re-renders

Reply:
changed to useMemo, thanks for the good suggestion